[KLUSH 이병재] add: users, products, orders, ratings app 추가 및 models.py 작성

redgpcmffld opened the pull request:
>wecode PR message template 및 체크 리스트 입니다. 
아래 사항들을 전부 작성/체크 하시고 PR 해주세요!

## 수정 사항 간략한 한줄 요약
-models.py 수정
		 
## 수정 사항들 자세한 내용
(여기에 수정 사항에 대한 자세한 내용을 작성해 주세요.)
-users app User class email max_length 300 => 255 변경, unique=true 로 하니 경고 메세지가 떠서 바꿨습니다.
-orders app Order class total_price, shipping_price IntegerField => DecimalField 변경
-orders app DiscountType class rate IntegerField => DecimalField 변경
-products app Product class price IntegerField => DecimalField 변경
-products app Option class weight, extra_cost IntegerField => DecimalField 변경
-ratings app Rate class rate IntegerField => DecimalField 변경
-users app nickname blank=true => null=true 변경
-orders app message blank=true => null=true 변경 

## 기타 질문 및 특이 사항
(궁금한 사항들, 하면서 느낀 점들 공유해주실 것이 있으면 자유롭게 작성해 주세요.)
-

## 체크 리스트 (아래 사항들이 전부 체크되어야만 merge가 됩니다!)
- [x] 필요한 test들을 완료하였고 기능이 제대로 실행되는지 확인 하였습니다.
- [x] Wecode의 코드 스타일 가이드에 맞추어 코드를 작성 하였습니다.
- [x] 제가 의도한 파일들과 수정 사항들만 커밋이 된 것을 확인 하였습니다.
- [x] 본 수정 사항들을 팀원들과 사전에 상의하였고 팀원들 모두 해당 PR에 대하여 알고 있습니다.

redgpcmffld committed:
models.py 완성

redgpcmffld committed:
fix: users app, ratings app, orders app 수정
-users app User class password 추가, nickname unique=True 삭제
-ratings app Rate class user ForeignKey on_delete 속성 변경
-orders app Order class order_status ForeignKey on_delete 속성 변경

redgpcmffld committed:
fix: migration reset

redsaumon commented on orders/models.py:
```diff
@@ -0,0 +1,50 @@
+class Order(models.Model):
+    user            = models.ForeignKey('users.User', on_delete=models.CASCADE)
+    order_status    = models.ForeignKey('OrderStatus', on_delete=models.PROTECT)
+    total_price     = models.IntegerField(null=True)
+    shipping_price  = models.IntegerField(null=True)
```
원화는 소수점을 사용하지 않지만  확장성을 고려해 DecimalField를 사용하시는 것을 추천합니다!

redgpcmffld replied:
알겠습니다!

redsaumon commented on orders/models.py:
```diff
@@ -0,0 +1,50 @@
+
+class ShippingInformation(models.Model):
+    address      = models.CharField(max_length=200)
+    message      = models.CharField(max_length=200, blank=True)
```
blank=True를 사용하신 이유가 있으신가요?(질문)

redgpcmffld replied:
배송메세지는 입력값이 없을수도 있는 속성인데 null=true로 하게 되면 데이터에서 빈 값을 조회할때 null인 것과 ''인 것 둘다 체크해야 하기 때문에 blank=true로 했습니다

redsaumon replied:
데이터에서 빈 값을 조회하려는 이유가 뭔가요?

redgpcmffld replied:
null=true로 변경했습니다

redsaumon commented on orders/models.py:
```diff
@@ -0,0 +1,50 @@
+
+class DiscountType(models.Model):
+    name  = models.CharField(max_length=45)
+    rate  = models.IntegerField()
```
가격이나 비율과 같은 값은 언제나 정수로 딱 맞게 맞아떨어지는 속성이라고 보기 어렵습니다. 정수로 보일지라도, 연산을 통해 실수가 될 수 있기 때문에 확장성으로 고려하여 DecimalField 등을 이용해 더 큰 범위로 설정해주시는 것이 좋습니다.

redgpcmffld replied:
바꾸겠습니다.

redsaumon commented on products/models.py:
```diff
@@ -0,0 +1,63 @@
+
+class Product(models.Model):
+    name          = models.CharField(max_length=45)
+    price         = models.IntegerField()
```
위에서 언급한 내용과 같은 이유로 수정 부탁드려요!

redsaumon commented on products/models.py:
```diff
@@ -0,0 +1,63 @@
+        db_table = "product_images"
+
+class ProductOption(models.Model):
+    weight     = models.IntegerField()
+    extra_cost = models.IntegerField()
```
확장성을 고려해 소수점까지 관리하시기를 추천드립니다!

redsaumon commented on ratings/models.py:
```diff
@@ -0,0 +1,26 @@
+from django.db import models
+
+class Rate(models.Model):
+    rate    = models.IntegerField()
```
별점이나 비율은 정수로 완벽하게 떨어지지 않을 경우가 많습니다.

redgpcmffld replied:
아 그 생각을 못 했네요 수정하겠습니다.

redsaumon commented on ratings/models.py:
```diff
@@ -0,0 +1,26 @@
+class Rate(models.Model):
+    rate    = models.IntegerField()
+    product = models.ForeignKey('products.Product', on_delete=models.CASCADE)
+    user    = models.ForeignKey('users.User', on_delete=models.SET_NULL, null=True)
```
user가 null일 때는 어떤 경우인가요?(질문)

redgpcmffld replied:
이미 제품에 별점을 주었던 유저가 회원탈퇴를 하는 경우입니다.

redsaumon replied:
👍

redsaumon commented on users/models.py:
```diff
@@ -0,0 +1,48 @@
+    phone_number = models.CharField(max_length=45, unique=True)
+    email        = models.EmailField(max_length=300, unique=True)
+    name         = models.CharField(max_length=45)
+    nickname     = models.CharField(max_length=45, blank=True)
```
blank=True가 무엇을 허용하는지 설명해주세요.

redgpcmffld replied:
null=true 처럼 db에 blank를 허용해주는건 줄 알고 썼는데 다시 찾아보니 is_valid로 유효성 검사를 할때 사용되는 거여서 지금 당장은 필요없는 속성값 같습니다. 제거해도 될까요?

redgpcmffld replied:
null=true로 변경했습니다

redgpcmffld committed:
fix: users app, ratings app, orders app, products app 수정
-적절하지 않은 datafield type 변경
-users.models.py User class_email 속성 max_length 변경

redgpcmffld committed:
fix: orders, users app 수정
-orders app Shipping class message blank=true => null=true 로 변경
-users app User class nickname blank=true => null=true 로 변경
